pkg/controller/install/webhook.go

```diff
@@ -200,69 +201,74 @@ func (i *StrategyDeploymentInstaller) createOrUpdateConversionWebhook(caPEM []by
-		// Conversion WebhookClientConfig needs to be set when Strategy is None
-		// https://v1-15.docs.kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definition-versioning/#configure-customresourcedefinition-to-use-conversion-webhooks
+			// Conversion WebhookClientConfig should not be set when Strategy is None
+			// https://v1-15.docs.kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definition-versioning/#specify-multiple-versions
+			// Conversion WebhookClientConfig needs to be set when Strategy is None
+			// https://v1-15.docs.kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definition-versioning/#configure-customresourcedefinition-to-use-conversion-webhooks
 
-		// use user defined path for CRD conversion webhook, else set default value
-		conversionWebhookPath := "/"
-		if desc.WebhookPath != nil {
-			conversionWebhookPath = *desc.WebhookPath
-		}
+			// use user defined path for CRD conversion webhook, else set default value
+			conversionWebhookPath := "/"
+			if desc.WebhookPath != nil {
+				conversionWebhookPath = *desc.WebhookPath
+			}
 
-		// Override Name, Namespace, and CABundle
-		crd.Spec.Conversion = &apiextensionsv1.CustomResourceConversion{
-			Strategy: "Webhook",
-			Webhook: &apiextensionsv1.WebhookConversion{
-				ClientConfig: &apiextensionsv1.WebhookClientConfig{
-					Service: &apiextensionsv1.ServiceReference{
-						Namespace: i.owner.GetNamespace(),
-						Name:      desc.DomainName() + "-service",
-						Path:      &conversionWebhookPath,
-						Port:      &desc.ContainerPort,
+			// Override Name, Namespace, and CABundle
+			crd.Spec.Conversion = &apiextensionsv1.CustomResourceConversion{
+				Strategy: "Webhook",
+				Webhook: &apiextensionsv1.WebhookConversion{
+					ClientConfig: &apiextensionsv1.WebhookClientConfig{
+						Service: &apiextensionsv1.ServiceReference{
+							Namespace: i.owner.GetNamespace(),
+							Name:      desc.DomainName() + "-service",
+							Path:      &conversionWebhookPath,
+							Port:      &desc.ContainerPort,
+						},
+						CABundle: caPEM,
 					},
-					CABundle: caPEM,
+					ConversionReviewVersions: desc.AdmissionReviewVersions,
 				},
-				ConversionReviewVersions: desc.AdmissionReviewVersions,
-			},
-		}
+			}
 
-		// update CRD conversion Specs
-		if _, err = i.strategyClient.GetOpClient().ApiextensionsInterface().ApiextensionsV1().CustomResourceDefinitions().Update(context.TODO(), crd, metav1.UpdateOptions{}); err != nil {
-			return fmt.Errorf("Error updating CRD with Conversion info: %v", err)
+			// update CRD conversion Specs
+			if _, err = i.strategyClient.GetOpClient().ApiextensionsInterface().ApiextensionsV1().CustomResourceDefinitions().Update(context.TODO(), crd, metav1.UpdateOptions{}); err != nil {
+				return fmt.Errorf("Error updating CRD with Conversion info: %w", err)
```

Comment:
Re-reading https://pkg.go.dev/k8s.io/client-go/util/retry#RetryOnConflict seems to imply that wrapping the error here might be problematic when identifying whether `IsConflict(err)`?

Reply:
`IsConflict` ends up calling `ReasonForError()`, which uses `errors.As` [here](https://github.com/kubernetes/apimachinery/blob/a644435e2c133d990b624858d9c01985d7f59adf/pkg/api/errors/errors.go#L650) to recursively unwrap and find the expected error type.

Reply:
ah

> If the format specifier includes a %w verb with an error operand, the returned error will implement an Unwrap method returning the operand. It is invalid to include more than one %w verb or to supply it with an operand that does not implement the error interface. 

interesting